src/libsystemd/sd-bus/bus-message.c

```diff
@@ -506,7 +506,9 @@ int bus_message_from_header(
                 m->fields_size = BUS_MESSAGE_BSWAP32(m, h->dbus1.fields_size);
                 m->body_size = BUS_MESSAGE_BSWAP32(m, h->dbus1.body_size);
 
-                if (sizeof(struct bus_header) + ALIGN8(m->fields_size) + m->body_size != message_size)
+                assert(message_size >= sizeof(struct bus_header));
```

Comment:
I'm not sure how likely it is this assertion can be triggered but given that those messages can be controlled by senders I think it would be better to avoid it. Bogus messages should be rejected earlier somewhere without crashing code using sd-bus to parse dbus messages.

Reply:
This is guaranteed by the conditions at the beginning of this function.